### PR TITLE
Add stream/task logs REST endpoints to API guide

### LINKED ISSUE #3322: Add RESTDoc for stream/task logs endpoints
Add Stream/Task logs REST endpoints as part of REST documentation.

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Add Stream/Task logs documentation tests
 - Update REST API guide with stream/task logs REST endpoints

Resolves #3322

### CODE REVIEW

**Comment by sabbyanandan** [comment]
@ghillert: Since you're going to be making a similar set of changes for https://github.com/spring-cloud/spring-cloud-dataflow/issues/3321, it'd be good if you can weigh-in on this - being on the same context would help.

File: spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc [hunk 1]
```diff
@@ -2643,3 +2644,118 @@ include::{snippets}/runtime-apps-documentation/list-all-apps/curl-request.adoc[]
...
+It is possible to get the application logs of the stream for the entire stream or a specific application inside the stream.
+The following topics provide more detail:
+
+* <<api-guide-resources-stream-logs-by-stream-name>>
+* <<api-guide-resources-stream-logs-by-app-name>>
+
+
+
+[[api-guide-resources-stream-logs-by-stream-name]]
+==== Get the applications' logs by the stream name
```
Comment (ghillert) [inline spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc]: Get the application logs by stream name

File: spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc [hunk 2]
```diff
@@ -2643,3 +2644,118 @@ include::{snippets}/runtime-apps-documentation/list-all-apps/curl-request.adoc[]
...
+
+To retrieve the logs of all the application of stream by the given stream name, query the `/streams/logs/<streamName>` endpoint by using `GET`.
```
Comment (ghillert) [inline spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc]: Use the HTTP `GET` method with the `/streams/logs/<streamName>` REST endpoint to retrieve all the application logs for the given stream name.

File: spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc [hunk 3]
```diff
@@ -2643,3 +2644,118 @@ include::{snippets}/runtime-apps-documentation/list-all-apps/curl-request.adoc[]
 
 include::{snippets}/runtime-apps-documentation/list-all-apps/http-response.adoc[]
 
+
+[[api-guide-resources-stream-logs]]
+=== Stream Logs
+
+It is possible to get the application logs of the stream for the entire stream or a specific application inside the stream.
+The following topics provide more detail:
```
Comment (ghillert) [inline spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc]: details

File: spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc [hunk 4]
```diff
@@ -2643,3 +2644,118 @@ include::{snippets}/runtime-apps-documentation/list-all-apps/curl-request.adoc[]
...
+* <<api-guide-resources-stream-logs-by-stream-name>>
+* <<api-guide-resources-stream-logs-by-app-name>>
+
+
+
+[[api-guide-resources-stream-logs-by-stream-name]]
+==== Get the applications' logs by the stream name
+
+To retrieve the logs of all the application of stream by the given stream name, query the `/streams/logs/<streamName>` endpoint by using `GET`.
+The following topics provide more detail:
```
Comment (ghillert) [inline spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc]: details

File: spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc [hunk 5]
```diff
@@ -2643,3 +2644,118 @@ include::{snippets}/runtime-apps-documentation/list-all-apps/curl-request.adoc[]
...
+[[api-guide-resources-stream-logs-by-stream-name-response-structure]]
+===== Response Structure
+
+include::{snippets}/stream-logs-documentation/get-logs-by-stream-name/http-response.adoc[]
+
+
+[[api-guide-resources-stream-logs-by-app-name]]
+==== Get the logs of a specific application from the stream
+
+To retrieve the logs of a specific application from the stream, query the `/streams/logs/<streamName>/<appName>` endpoint by using `GET`.
```
Comment (ghillert) [inline spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc]: ... endpoint using the `GET` HTTP method.

File: spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc [hunk 6]
```diff
@@ -2643,3 +2644,118 @@ include::{snippets}/runtime-apps-documentation/list-all-apps/curl-request.adoc[]
...
+include::{snippets}/stream-logs-documentation/get-logs-by-app-name/http-response.adoc[]
+
+
+
+[[api-guide-resources-task-logs]]
+=== Task Logs
+
+It is possible to get the task execution log of a specific task execution.
+
+The following topics provide more detail:
```
Comment (ghillert) [inline spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc]: topic provides more details

File: spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc [hunk 7]
```diff
@@ -2643,3 +2644,118 @@ include::{snippets}/runtime-apps-documentation/list-all-apps/curl-request.adoc[]
...
+===== Response Structure
+
+include::{snippets}/stream-logs-documentation/get-logs-by-app-name/http-response.adoc[]
+
+
+
+[[api-guide-resources-task-logs]]
+=== Task Logs
+
+It is possible to get the task execution log of a specific task execution.
```
Comment (ghillert) [inline spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc]: for a specific task execution.

File: spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc (hunk 1 above)
Comment (ilayaperumalg) [reply spring-cloud-dataflow-docs/src/main/asciidoc/api-guide.adoc]: I intend to use `applications'` to mean multiple applications logs :-)

**Comment by ghillert** [comment]
LGTM - rebased and merged with commit 1ec1ff3c19c3c5a743215da7914da4a2a0f59381

**Pull request closed without merging**